exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/scan/file/FileMetadata.java

```diff
@@ -31,9 +32,15 @@
 
   private final Path filePath;
   private final String[] dirPath;
+  private final FileSystem fs;
 
   public FileMetadata(Path filePath, Path selectionRoot) {
+    this(filePath, selectionRoot, null);
+  }
+
+  public FileMetadata(Path filePath, Path selectionRoot, FileSystem fs) {
```

Comment:
Can we add a comment to relate these? The `filePath` must be fully qualified for the file system `fs`. The `selectionRoot`, if provided, must be a prefix of `filePath`. `filePath` should not include the file system prefix. (I *think* those are the relationships, but your understanding is fresh, is this correct?)

Reply:
This class has good JavaDoc which explains the relation between `filePath` and `selectionRoot`. The file system should accept paths with or without file system prefix, so no additional relation is introduced.